libraries/CurieBLE/src/internal/BLEServiceImp.cpp

```diff
@@ -347,44 +370,143 @@ uint8_t BLEServiceImp::discoverResponseProc(bt_conn_t *conn,
         }
     }
     
-    pr_debug(LOG_MODULE_BLE, "%s-%d:ret-%d",__FUNCTION__, __LINE__, retVal);
+    //pr_debug(LOG_MODULE_BLE, "%s-%d:ret-%d",__FUNCTION__, __LINE__, retVal);
     if (retVal == BT_GATT_ITER_STOP)
     {
         if (errno == ENOMEM)
         {
             _cur_discover_chrc = NULL;
             return retVal;
         }
-        const BLECharacteristicLinkNodeHeader* chrcHeader = &_characteristics_header;
-        BLECharacteristicImp* chrcCurImp = NULL;
-        BLECharacteristicNodePtr node = chrcHeader->next;
         
-        pr_debug(LOG_MODULE_BLE, "%s-%d: node-%p",__FUNCTION__, __LINE__, node);
-        // Discover next service
-        while (node != NULL)
+        if (false == _reading)
         {
-            chrcCurImp = node->value;
-            
-            if (NULL == _cur_discover_chrc)
+            discoverNextCharacteristic(device);
+        }
+    }
+    return retVal;
+}
+
+void BLEServiceImp::discoverNextCharacteristic(BLEDevice &bledevice)
+{
+    const BLECharacteristicLinkNodeHeader* chrcHeader = &_characteristics_header;
+    BLECharacteristicImp* chrcCurImp = NULL;
+    BLECharacteristicNodePtr node = chrcHeader->next;
+    
+    //pr_debug(LOG_MODULE_BLE, "%s-%d: node-%p",__FUNCTION__, __LINE__, node);
```

Comment:
commented pr_debug? I forgot what our policy is on this. Personally I'd take it out.